### DISCUSSION TITLE
How to add components or utilities from separate files?

### DISCUSSION THREAD

**Discussion by ningxie1991** [tailwindlabs/tailwindcss#10736] [Help]
Hello,

I am trying to build a react component library using Tailwindcss for styling management. I would like to extract reusable styles into components or classes. I understand we can do it directly in the index.css file: 

![image](https://user-images.githubusercontent.com/3387698/223389306-79e9b7c1-03e0-4ab9-a7ad-c18d92929fc9.png)
 
or add to the utilities like so:
![image](https://user-images.githubusercontent.com/3387698/223389490-dfe56121-fd30-418d-836e-4a04b82af25d.png)

However, if I have many components, classes or utilities, I want to keep them in separate css files and import in the tailwindcss.config.js, kind of like what daisyUI is doing [here](https://github.com/saadeghi/daisyui/tree/master/src/components/styled). I have tried but could not get the separate css files to load with tailwind. 

I am using npm, rollup and postcss to package the component library. 

My attempt was like this:

|-- src
|    |-- components
|    |    |-- Navbar.tsx
|    |-- styles
|    |    |-- components.css
|-- tailwindcss.config.js

```
// Navbar.tsx
...
<div className="navbar> </>
...
```
```
// components.css
.navbar {
    @apply w-full h-16 flex items-center;
}
```
```
// tailwindcss.config.js
   plugins: [
        plugin(function ({ addUtilities }) {
            addUtilities(require("./src/styles/components.css"));
        })
    ],
```
The approach above was not working. Does anyone know how to do that?

Thanks!!

**Comment by wongjn** [comment]
You could use [build-time imports](https://tailwindcss.com/docs/using-with-preprocessors#build-time-imports) in your main CSS file.
